bin/truffleruby.sh

```diff
@@ -72,7 +72,13 @@ java_args+=("-Xbootclasspath/a:$truffle/dists/truffle-api.jar:$graal_sdk")
 CP="$CP:$truffle/dists/truffle-nfi.jar"
 CP="$CP:$root/mxbuild/dists/truffleruby-launcher.jar"
 CP="$CP:$root/mxbuild/dists/truffleruby.jar"
-java_args+=("-Dtruffle.nfi.library=$truffle/truffle-nfi-native/bin/libtrufflenfi.$(libext)")
+
+# find libtrufflenfi
+libtrufflenfi=()
+# look in architecture specific mxbuild dir then look to common mxbuild dir
+libtrufflenfi+=($(ls -1 $truffle/*/truffle-nfi-native/bin/libtrufflenfi.$(libext) $truffle/truffle-nfi-native/bin/libtrufflenfi.$(libext) 2> /dev/null || true))
+libtrufflenfi=${libtrufflenfi[0]} # take first candidate
+java_args+=("-Dtruffle.nfi.library=$libtrufflenfi")
```

Comment:
Why do we need this?
Is it a Sulong change or a mx change?

Reply:
mx now does this architecture-specific build directory thing.

Reply:
Can we simplify to just use the new format and require a recent mx to build TruffleRuby then?

Reply:
Also, `echo` should be enough here to resolve the `*`, no need for `ls`.

Reply:
We should also update Truffle at the same time, so that whether we use a binary or source Truffle it's consistently the new layout.

Reply:
Note that this solution will not work when you're sharing the workspace between platforms (e.g. shared folder for VMs), which was the original motivation for the change in mx.

Can't we just do something similar to https://github.com/graalvm/truffleruby/pull/691, but instead of generating the included file with a shell script, use mx to generate it? Then you don't need to hardcode paths anywhere, and are not affected by future mx changes at all...

Reply:
Let's merge #691 and reuse a similar mechanism then.
We can have the `TruffleRubyLauncherBuildTask` generates a `truffleruby_env` which contains the path to libtrufflenfi.

Reply:
@eregon It's easier to do this in 2 steps, then trying to update everything at once. Echo does not print a file per line.

@rschatz right this is broken, I've pushed new version.

Reply:
How should truffleruby_env approach work? Afaik, It will be in the same shared workspace and picked up by different VMs.

Reply:
That's mostly fine if `mx build` always re-generates truffleruby_env. And anyway TruffleRuby doesn't support yet a single clone used by multiple platforms, that's a much bigger change.